host.go

```diff
@@ -360,10 +361,17 @@ func (h *Host) Create(name string) error {
 		return err
 	}
 
+	// create the instance
 	if err := h.Driver.Create(); err != nil {
 		return err
 	}
 
+	// install docker
+	if err := h.Provision(); err != nil {
```

Comment:
I am +1 on this - originally the EC2 driver had something similar but I think it was removed.

I'd be a fan of checking to see if Docker's already installed as well (maybe we could provide the drivers with a function that checks for the presence of the Docker socket).